everestrs/everestrs/src/lib.rs

```diff
@@ -0,0 +1,196 @@
+    //    after it. Rust drops fields in declaration order, hence `cpp_module` should come before
+    //    `module_impl` in this struct.
+    cpp_module: cxx::UniquePtr<ffi::Module>,
+    module_impl: Pin<Box<T>>,
+}
+
+impl<T: GenericModule> Runtime<T> {
+    fn ptr_to_module_impl(&self) -> *const c_char {
+        &*self.module_impl.as_ref() as *const T as *const c_char
+    }
+
+    // TODO(hrapp): This function could use some error handling.
+    pub fn from_commandline(module_impl: T) -> Self {
+        let args: Args = argh::from_env();
+        let mut cpp_module = ffi::create_module(
+            &args.module,
+            &args.prefix.to_string_lossy(),
+            &args.conf.to_string_lossy(),
+        );
+        let manifest_json = cpp_module.as_mut().unwrap().initialize();
+        let manifest: schema::Manifest = manifest_json.deserialize();
+        let module = Self {
+            cpp_module,
+            module_impl: Box::pin(module_impl),
+        };
+
+        // Implement all commands for all of our implementations, dispatch everything to the
+        // GenericModule.
+        for (implementation_id, provides) in manifest.provides {
+            let interface_s = module.cpp_module.get_interface(&provides.interface);
+            let interface: schema::Interface = interface_s.deserialize();
+            for (name, _) in interface.cmds {
+                let meta = ffi::CommandMeta {
+                    implementation_id: implementation_id.clone(),
+                    name,
+                    obj: module.ptr_to_module_impl(),
+                };
+                unsafe {
+                    module.cpp_module.as_ref().unwrap().provide_command(
+                        meta,
+                        |meta: &ffi::CommandMeta, params: ffi::JsonBlob| {
+                            let module_impl = &mut *(meta.obj as *mut T);
+                            let out = match module_impl.handle_command(
+                                &meta.implementation_id,
+                                &meta.name,
+                                params.deserialize(),
+                            ) {
+                                Err(e) => panic!("Error calling command: {e:?}"),
+                                Ok(out) => out,
+                            };
+                            ffi::JsonBlob::from_vec(serde_json::to_vec(&out).unwrap())
+                        },
+                    );
+                }
+            }
+        }
+
+        // Since users can choose to overwrite `on_ready`, we can call signal_ready right away.
+        unsafe {
+            module
+                .cpp_module
+                .as_ref()
+                .unwrap()
+                .signal_ready(module.ptr_to_module_impl(), |obj| {
+                    let module_impl = &mut *(obj as *mut T);
+                    module_impl.on_ready();
+                });
+        }
```

Comment:
My thought process here was the following: 

- Since the Rust module can do any prep before it initially EVerest, it is ready for EVerest as soon as it instantiates it.
- It is a non-goal for the Rust code to manually deal with fullfillments, this should all be transparently done by the framework. The flexibility that for example Python and JS is giving is currently explicitly not desired - it leaves too much room for human error. 
- It might or might not be interesting to the module to know when every module is ready, so I added the `on_ready` callback for it too know.

I might miss an important point here, but if so, I do not see it now. I added a TODO to discuss this at a later point in time.